tests/models/gpt_bigcode/test_modeling_gpt_bigcode.py

```diff
@@ -0,0 +1,874 @@
+        model.to(torch_device)
+        tokenizer = GPT2TokenizerFast.from_pretrained("bigcode/santacoder")
+
+        tokenizer.padding_side = "left"
+
+        # This tokenizer has no pad token, so we have to set it in some way
+        # Define PAD Token = EOS Token = 50256
+        tokenizer.pad_token = tokenizer.eos_token
+        model.config.pad_token_id = model.config.eos_token_id
+
+        # use different length sentences to test batching
+        sentences = [
+            "def hello_world():",
+            "const int x = 5;",
+        ]
+
+        inputs = tokenizer(sentences, return_tensors="pt", padding=True)
+        input_ids = inputs["input_ids"].to(torch_device)
+        token_type_ids = torch.cat(
+            [
+                input_ids.new_full((input_ids.shape[0], input_ids.shape[1] - 1), 0),
+                input_ids.new_full((input_ids.shape[0], 1), 500),
+            ],
+            dim=-1,
+        )
+
+        outputs = model.generate(
+            input_ids=input_ids,
+            attention_mask=inputs["attention_mask"].to(torch_device),
+        )
+
+        outputs_tt = model.generate(
+            input_ids=input_ids,
+            attention_mask=inputs["attention_mask"].to(torch_device),
+            token_type_ids=token_type_ids,
+        )
+
+        inputs_non_padded = tokenizer(sentences[0], return_tensors="pt").input_ids.to(torch_device)
+        output_non_padded = model.generate(input_ids=inputs_non_padded)
+
+        num_paddings = inputs_non_padded.shape[-1] - inputs["attention_mask"][-1].long().sum().cpu().item()
+        inputs_padded = tokenizer(sentences[1], return_tensors="pt").input_ids.to(torch_device)
+        output_padded = model.generate(input_ids=inputs_padded, max_length=model.config.max_length - num_paddings)
+
+        batch_out_sentence = tokenizer.batch_decode(outputs, skip_special_tokens=True)
+        batch_out_sentence_tt = tokenizer.batch_decode(outputs_tt, skip_special_tokens=True)
+        non_padded_sentence = tokenizer.decode(output_non_padded[0], skip_special_tokens=True)
+        padded_sentence = tokenizer.decode(output_padded[0], skip_special_tokens=True)
+
+        expected_output_sentence = [
+            "Hello, my dog is a little bit of a mess. I'm not sure if he's going",
+            "Today, I'm going to be doing a lot of research on this. I",
+        ]
+        self.assertListEqual(expected_output_sentence, batch_out_sentence)
+        self.assertTrue(batch_out_sentence_tt != batch_out_sentence)  # token_type_ids should change output
+        self.assertListEqual(expected_output_sentence, [non_padded_sentence, padded_sentence])
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_name in GPT_BIGCODE_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
+            model = GPTBigCodeModel.from_pretrained(model_name)
+            self.assertIsNotNone(model)
+
+
+# Skip these as the model does not support scale_attn_by_inverse_layer_idx - irrelevant
```

Comment:
@younesbelkada Isn't `scale_attn_by_inverse_layer_idx` just for a small subset of this class?

Reply:
I don't think we should add this test suite, `scale_attn_by_inverse_layer_idx` is replaced by `scale_attention_softmax_in_fp32`, also I added some integration tests that uses the model on the Hub that have: https://huggingface.co/bigcode/gpt_bigcode-santacoder/blob/main/config.json#L22 
The test tests simple generation + batched generation and should cover the possible use cases of the model and if in the future there will be PRs that will break `scale_attention_softmax_in_fp32` feature, we'll notice it with that test. If you are happy with this explanation we should remove the test that I have commented!

Reply:
Removed! Beam search etc are tested through `GenerationMixin` superclass